nexus/social_media/services.py

```diff
@@ -4,16 +4,167 @@
+                                    headers=headers)
+    if response_assets.status_code == status.HTTP_200_OK:
+        json_response_assets = response_assets.json()
+
+        value = json_response_assets.get('value')
+        upload_mechanism = value.get('uploadMechanism')
+        upload_http_request = upload_mechanism.get('com.linkedin.digitalmedia.uploading.MediaUploadHttpRequest')
+        upload_url = upload_http_request.get('uploadUrl')
+
+        asset = value.get('asset')
+
+        response_upload_image = requests.post(upload_url,
+                                              data=post.image.file.read(),
+                                              headers={'Authorization': f"Bearer {linkedin_auth['access_token']}"})
+        if response_upload_image.status_code == status.HTTP_201_CREATED:
+            return asset
+        else:
+            appropriate_response_action(response_upload_image)
+    else:
+        appropriate_response_action(response_assets)
+
+
+def publish_on_linkedin(post_id):
+    """Function to post on linkedin
+
+    :param post_id: UUID of the post instance to be posted.
+
+    :returns: A valid HTTP response
+
+    """
+    linkedin_auth = settings.LINKEDIN_AUTH
+    author = f"urn:li:organization:{linkedin_auth['organization_id']}"
+
+    headers = {'X-Restli-Protocol-Version': '2.0.0',
+               'Content-Type': 'application/json',
+               'Authorization': f"Bearer {linkedin_auth['access_token']}"}
+
+    api_url_ugc = f'{settings.LINKEDIN_API_URL_BASE}ugcPosts'
+
+    post = Post.objects.get(pk=post_id)
+
+    post_data = {
+        'author': author,
+        'lifecycleState': 'PUBLISHED',
+        'specificContent': {
+            'com.linkedin.ugc.ShareContent': {
+                'shareCommentary': {
+                    'text': post.text
+                },
+                'shareMediaCategory': 'NONE'
+            },
+        },
+        'visibility': {
+            'com.linkedin.ugc.MemberNetworkVisibility': 'PUBLIC'
+        },
+    }
+
+    if post.image:
+        asset = upload_image_to_linkedin(author, headers, post, linkedin_auth)
+
+        specific_content = post_data.get('specificContent')
+        share_content = specific_content.get('com.linkedin.ugc.ShareContent')
+
+        image_media = [{
+            'status': 'READY',
+            'description': {
+                'text': ''
+            },
+            'media': asset,
+            'title': {
+                'text': ''
+            }
+        }]
+        share_content.update(shareMediaCategory='IMAGE',
+                             media=image_media)
+        image_response = requests.post(api_url_ugc, headers=headers,
+                                       json=post_data)
+        return image_response
+    elif post.text:
+        response = requests.post(api_url_ugc, headers=headers, json=post_data)
+        return response
+
+
+def appropriate_response_action(response):
```

Comment:
Please raise errors directly. The entire purpose of renaming status codes as  `status.HTTP_400_BAD_REQUEST` was to be clear & precise.

This function adds more level of abstraction which is not needed.

Reply:
Re-reading the entire code again. I think we'll just rename this function as `check_and_raise_error_from_linkedin_response`